### ISSUE TITLE
Find a better solution for when no pattern is detected by an RGB camera in a collection

### ISSUE THREAD

**Issue by manuelgitgomes** [lardemua/atom#765]
This is regarding this block of code:

https://github.com/lardemua/atom/blob/711ac784be2a59816e6697c0ffaaf5a4c72638a3/atom_calibration/src/atom_calibration/collect/patterns.py#L376-L378

We should really implement this, because we are locking ATOM so that every collection has at least a single collection.
I suggest looking at the previous collection and copying the transformation.
If no previous collection exists, then randomize.

What do you think @miguelriemoliveira?

**Comment by miguelriemoliveira** [comment]
> We should really implement this, because we are locking ATOM so that every collection has at least a single collection.

I don't understand ... Can you rephrase?

> I suggest looking at the previous collection and copying the transformation.
> If no previous collection exists, then randomize.

Ok, makes sense. Its better than nothing. When you have let me know so I can take a look.

**Comment by manuelgitgomes** [comment]
> I don't understand ... Can you rephrase?

Sorry, my bad.
I meant to say:

> We should really implement this, because we are locking ATOM so that every collection has at least a single rgb detection.

**Comment by miguelriemoliveira** [comment]
> We should really implement this, because we are locking ATOM so that every collection has at least a single rgb detection.

Right, its a bummer.

**Comment by manuelgitgomes** [comment]
Sorry @miguelriemoliveira, isn't there a possibility of just saying "detected: false"?
That way, if the pattern is not fixed, this collection is not accounted for, as was previously, right?

**Comment by miguelriemoliveira** [comment]
Hi @manuelgitgomes ,

I don't think that's a possibility. If we do not have any rgb camera in a system, we would say detected: false for all collections and none of them would be used for calibration. That way we couldn't calibrate.

We will need to initialize the pose of the pattern with some numbers, since we need to estimate the pose of the pattern to carry out the optimization. We can discard some collections by saying the pattern was not detected, but we can't discard them all.

> That way, if the pattern is not fixed, this collection is not accounted for, as was previously, right?

But even with a fixed pattern we need to estimate its pose. It is only one pose for all collections, but we still need to estimate it, and we still need to initialize the pose to start the optimization.

**Comment by miguelriemoliveira** [comment]
After #789 I have been thinking about this.

Currently what we force that, for all collections, a pattern must be found by at least one rgb sensor.

This has limiting implications:

1. All robotic systems in atom require at least one rgb sensor
2. All collections must have the pattern in such position that the rgb sensor is able to detect it

This is probably a result the fact that before we had the detection of the pattern pose carried out during calibration, and recently decided to move this functionality to the collect data phase (#741).

Right now what we do in case the pattern is not detected by an rgb sensor is to throw an atom error 

https://github.com/lardemua/atom/blob/85b0e6755f4a625aad48c036cc2fb912d684f5a5/atom_calibration/src/atom_calibration/collect/patterns.py#L453-L460

I think @manuelgitgomes suggestion from above makes some sense. The idea is that if we do not detect the pattern in that collection, we record the dataset collection anyway and throw the problem forward.

The lack of a detection can be later tackled either by using the dataset playback to manually set an estimate of the pattern's pose, or in the calibrate script by saying we do not use this collection as we cannot establish the initial pose of the pattern.

Before I disagreed with @manuelgitgomes , but I was thinking about all collections at once. If this only happens to one or another collection, I think it would be ok.

So the decision is to remove the atom error and save the collection anyway, signaling that the pattern is not detected by any sensor.

**Comment by miguelriemoliveira** [comment]
This is implemented and I tested a calibration where this occurs. Calibrate handles it well. Closing.
